### PR TITLE
[JENKINS-56109] Compatibility with table => div

### PR DESCRIPTION
**Pull request by timja**
https://github.com/jenkinsci/jenkins/pull/3895/files#r422546234

cc @daniel-beck 

Testing:
* test with: https://github.com/jenkinsci/jenkins/pull/3895 `docker run --rm -ti -p 8080:8080 -e ID=3895 jenkins/core-pr-tester`
* Set 'Project-based Matrix Authorization Strategy'
* Go to the `configure` page and make sure the 'Enable node-based security' optional property behaves properly

### CODE REVIEW

**Comment by daniel-beck** [comment]
Is this PR complete? There are two other occurrences (folder and node properties).

**Comment by timja** [comment]
> Is this PR complete? There are two other occurrences (folder and node properties).

Should have searched better -.-, added to others

**Comment by res0nance** [comment]
Maybe add a `TODO` so this workaround will be removed when its no longer needed?

**Comment by timja** [comment]
> Maybe add a `TODO` so this workaround will be removed when its no longer needed?

thoughts on naming? hard to put a version when we don't know when it will be shipped

@daniel-beck is this waiting on anything?

**Comment by daniel-beck** [comment]
> thoughts on naming? hard to put a version when we don't know when it will be shipped

`TODO remove this once https://github.com/jenkinsci/jenkins/pull/3895 is in the core baseline`

File: src/main/resources/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty/config.groovy [hunk 1]
```diff
@@ -8,9 +8,21 @@ def st = namespace("jelly:stapler")
...
+        blockWrapper {
             f.dropdownDescriptorSelector(title: _("Inheritance Strategy"), descriptors: InheritanceStrategyDescriptor.getApplicableDescriptors(my.class), field: 'inheritanceStrategy')
             st.include(class: "hudson.security.GlobalMatrixAuthorizationStrategy", page: "config")
         }
     }
-}
\ No newline at end of file
+}
+
+def blockWrapper(Closure closure) {
```
Comment (timja) [inline src/main/resources/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty/config.groovy:18]:
```suggestion
// TODO remove this once https://github.com/jenkinsci/jenkins/pull/3895 is in the core baseline
def blockWrapper(Closure closure) {
```

File: src/main/resources/hudson/security/AuthorizationMatrixProperty/config.groovy [hunk 2]
```diff
@@ -8,9 +8,21 @@ def st = namespace("jelly:stapler")
...
+        blockWrapper {
             f.dropdownDescriptorSelector(title: _("Inheritance Strategy"), descriptors: InheritanceStrategyDescriptor.getApplicableDescriptors(my.class), field: 'inheritanceStrategy')
             st.include(class: "hudson.security.GlobalMatrixAuthorizationStrategy", page: "config")
         }
     }
-}
\ No newline at end of file
+}
+
+def blockWrapper(Closure closure) {
```
Comment (timja) [inline src/main/resources/hudson/security/AuthorizationMatrixProperty/config.groovy:18]:
```suggestion
// TODO remove this once https://github.com/jenkinsci/jenkins/pull/3895 is in the core baseline
def blockWrapper(Closure closure) {
```

File: src/main/resources/org/jenkinsci/plugins/matrixauth/AuthorizationMatrixNodeProperty/config.groovy [hunk 3]
```diff
@@ -7,8 +7,21 @@ def f = namespace(FormTagLib)
...
 f.nested {
-    table(style: "width: 100%") {
+    blockWrapper {
         f.dropdownDescriptorSelector(title: _("Inheritance Strategy"), descriptors: InheritanceStrategyDescriptor.getApplicableDescriptors(my?.class?:hudson.model.Node.class), field: 'inheritanceStrategy')
         st.include(class: "hudson.security.GlobalMatrixAuthorizationStrategy", page: "config")
     }
+    
+}
+
+def blockWrapper(Closure closure) {
```
Comment (timja) [inline src/main/resources/org/jenkinsci/plugins/matrixauth/AuthorizationMatrixNodeProperty/config.groovy:17]:
```suggestion
// TODO remove this once https://github.com/jenkinsci/jenkins/pull/3895 is in the core baseline
def blockWrapper(Closure closure) {
```

**Comment by daniel-beck** [comment]
> is this waiting on anything?

Me to have time. Recent PRs caused regressions, so I would rather not yolo this without a lot of testing.

**Comment by fqueiruga** [comment]
Once you get to it can you please look into the inline JS on https://github.com/jenkinsci/matrix-auth-plugin/blob/master/src/main/resources/hudson/security/GlobalMatrixAuthorizationStrategy/config.jelly to verify that it is not affected?

**Comment by daniel-beck** [comment]
Looks like `buildPlugin.recommendedConfigurations()` broke some time between May 10 and June 7 for a plugin with a recent baseline 😞

**Comment by timja** [comment]
It was raised to a recent one and then reverted :( 

Lots of plugin builds broke...

**Comment by daniel-beck** [comment]
Removed in https://github.com/jenkinsci/matrix-auth-plugin/commit/9616b3877395aa1e8b0b403601a968fa7a461193 for this plugin.

2.164.x is an unreasonably old choice, the update center doesn't even support that anymore.

**Comment by timja** [comment]
Closing in favour of https://github.com/jenkinsci/matrix-auth-plugin/pull/86

**Pull request closed without merging**